exporters/otlp/src/otlp_http_client.cc

```diff
@@ -904,9 +898,11 @@ OtlpHttpClient::createSession(
       {
         std::cerr << error_message << '\n';
       }
-      OTEL_INTERNAL_LOG_ERROR(error_message.c_str());
+      OTEL_INTERNAL_LOG_ERROR(error_message);
 
-      return opentelemetry::sdk::common::ExportResult::kFailure;
+      const auto result = opentelemetry::sdk::common::ExportResult::kFailure;
+      result_callback(result);
```

Comment:
Here and in other places, the call to result_callback() has moved a bit.

Could you provide more details ?

Reply:
Absolutely. The scan reports an _[accessMoved]_ warning [here](https://github.com/open-telemetry/opentelemetry-cpp/actions/runs/11948772032/job/33306927417#step:4:440)

This is due to the move that happens on higher `OtlpHttpClient::Export()` [L753](https://github.com/open-telemetry/opentelemetry-cpp/pull/3150/files/e9b225295e05a2baf71708e6486036042ed69542#diff-eb7f99ebc152d9cc0f4781cc0a0388907e4d4b58271dba85eba99fd16b880674R753) which looks like it is likely undefined behavior calling it from that point on.

Since the **_moved from_** result_callback appears to be used only to consume the result from `createSession()`, my solution was to move the invocation closer to the call sites where it's been **_moved to_**.

Reply:
Good call on `result_callback`, thanks for finding and fixing this.